src/peers/WorldMap/WorldMap.js

```diff
@@ -1,44 +1,112 @@
+    )
+  }, [doSetSelectedPeers])
+
+  const handlePopoverMouseEnter = useCallback(() => clearTimeout(selectedTimeout), [selectedTimeout])
+
   return (
-    <div className={`relative ${className}`}>
-      <div className='mb4 overflow-hidden flex flex-column items-center'>
-        <div style={{ width, height, background: `transparent url(${staticMapSrc}) center no-repeat`, backgroundSize: 'auto 100%' }}>
-          <GeoPath width={width} height={height}>
-            { ({ path }) => (
-              <MapPins width={width} height={height} path={path} />
-            )}
-          </GeoPath>
+    <div className="flex flex-column">
+      <div className={`relative ${className}`}>
+        <div className='mb1 flex flex-column items-center'>
+          <div className="relative" style={{ width, height, background: `transparent url(${staticMapSrc}) center no-repeat`, backgroundSize: 'auto 100%' }}>
+            <GeoPath width={width} height={height}>
+              { ({ path }) => (
+                <MapPins width={width} height={height} path={path} handleMouseEnter={ handleMapPinMouseEnter } handleMouseLeave= { handleMapPinMouseLeave } />
+              )}
+            </GeoPath>
+            { selectedPeers?.peerIds && (
+              <Popover show={ !!(selectedPeers.top && selectedPeers.left) } top={ selectedPeers.top } left={ selectedPeers.left } align='top'
+                handleMouseEnter={ handlePopoverMouseEnter } handleMouseLeave={ handleMapPinMouseLeave }>
+                <PeerInfo ids={ selectedPeers.peerIds } t={t}/>
+              </Popover>)
+            }
+          </div>
+        </div>
+        <div className='mapFooter absolute bottom-0 left-0 right-0'>
+          <div className='flex flex-auto flex-column items-center self-end pb5-ns no-select'>
+            <div className='f1 fw5 black'><PeersCount /></div>
+            <div className='f4 b ttu charcoal-muted'>{t('peers')}</div>
+          </div>
         </div>
       </div>
-      <div className='absolute bottom-0 left-0 right-0'>
-        <div className='flex flex-auto flex-column items-center self-end pb5-ns no-select'>
-          <div className='f1 fw5 black'><PeersCount /></div>
-          <div className='f4 b ttu charcoal-muted'>{t('peers')}</div>
+      <div className='relative flex justify-end pt2 pb4'>
+        <div className='f6 p2 no-select flex items-center'>
+          <span className='f6 charcoal-muted pr3'>{t('index')}: </span>
+          <i className='mapDotExplanation mr1' style={{ width: getDotsSize(1) * 2, height: getDotsSize(1) * 2, backgroundColor: getDotsColor(1) }}></i>1-10 {t('peers')}
+          <i className='mapDotExplanation ml3 mr1' style={{ width: getDotsSize(50) * 2, height: getDotsSize(50) * 2, backgroundColor: getDotsColor(50) }}></i> 10-100 {t('peers')}
+          <i className='mapDotExplanation ml3 mr1' style={{ width: getDotsSize(110) * 2, height: getDotsSize(110) * 2, backgroundColor: getDotsColor(110) }}></i>100+ {t('peers')}
```

Comment:
> ![2020-04-24--03-04-25](https://user-images.githubusercontent.com/157609/80164214-a8399e80-85d8-11ea-91de-592037878d4d.png)

i18n nit: having "Peers" multiple times may look bad in languages where the world is long etc.

I believe it will be safer to simplify this something like:

> ![2020-04-24--03-05-21](https://user-images.githubusercontent.com/157609/80164230-b1c30680-85d8-11ea-8928-7985ea5f0cdd.png)

This will  also enable us to add one more level if we want (eg. 1-10, 10-50, 50-100, 100+)